keanu-python/build.gradle

```diff
@@ -57,9 +57,37 @@ task pytest {
     }
 }
 
+task generatePythonDocumentation {
+    group "pythonDocumentation"
+    doLast {
+        exec {
+            workingDir("../docs/")
+            commandLine "rm", "-rf", "python/"
+        }
+        exec {
+            workingDir("docs/")
+            commandLine "pipenv", "run", "make", "html"
+        }
+        exec {
+            workingDir("docs/")
+            commandLine "pipenv", "run", 'python', 'remove_underscores.py'
+        }
+        exec {
+            commandLine 'mv', 'docs/_build/html/', '../docs/python/'
+        }
+        exec {
+            workingDir("docs/")
+            commandLine 'rm', '-rf', '_build/'
+        }
+    }
+}
+
 test.dependsOn(pytest)
 pytest.dependsOn(installPipenv)
 pytest.dependsOn(pythonVersionInfo)
 pytest.dependsOn(installDependencies)
 pythonVersionInfo.mustRunAfter(installPipenv)
 installDependencies.mustRunAfter(pythonVersionInfo)
+generatePythonDocumentation.dependsOn(":compileJava")
+generatePythonDocumentation.dependsOn(installPipenv)
+generatePythonDocumentation.dependsOn(installDependencies)
```

Comment:
The task dependency logic is getting quite confusing in this file. We can consolidate it to this:

    pythonVersionInfo.mustRunAfter(installPipenv)
    installDependencies.mustRunAfter(pythonVersionInfo)
    
    task preparePythonEnvironment {
        dependsOn(installPipenv)
        dependsOn(pythonVersionInfo)
        dependsOn(installDependencies)
        dependsOn(":keanu-project:preparePythonClasspath")
        dependsOn(":codegen:codeGen")
    }
    
    generatePythonDocumentation.dependsOn(preparePythonEnvironment)
    pytest.dependsOn(preparePythonEnvironment)
    test.dependsOn(pytest)

You can then delete lines 85-93 and lines 1-2.

You can check that the dependencies are still as you intend them by using the `--dry-run` argument, to print them out, e.g. 

    ./gradlew :keanu-python:generatePythonDocumentation --dry-run